contracts/src/factory/facets/architect/ArchitectBase.sol

```diff
@@ -338,3 +346,195 @@ abstract contract ArchitectBase is Factory, IArchitectBase {
     );
   }
 }
+
+abstract contract ArchitectBaseV2 is ArchitectBase {
+  function _createSpaceV2(
+    SpaceInfoV2 memory spaceInfo
+  ) internal returns (address spaceAddress) {
+    ArchitectStorage.Layout storage ds = ArchitectStorage.layout();
+    ImplementationStorage.Layout storage ims = ImplementationStorage.layout();
+
+    // get the token id of the next space
+    uint256 spaceTokenId = ims.spaceToken.nextTokenId();
+
+    // deploy space
+    spaceAddress = _deploySpaceV2(spaceTokenId, spaceInfo.membership);
+
+    // save space info to storage
+    ds.spaceCount++;
+
+    // save to mappings
+    ds.spaceByTokenId[spaceTokenId] = spaceAddress;
+    ds.tokenIdBySpace[spaceAddress] = spaceTokenId;
+
+    // mint token to and transfer to Architect
+    ims.spaceToken.mintSpace(
+      spaceInfo.name,
+      spaceInfo.uri,
+      spaceAddress,
+      spaceInfo.shortDescription,
+      spaceInfo.longDescription
+    );
+
+    // deploy user entitlement
+    IUserEntitlement userEntitlement = IUserEntitlement(
+      _deployEntitlement(ims.userEntitlement, spaceAddress)
+    );
+
+    // deploy token entitlement (Assume the implementation is an IRuleEntitlementV2)
+    IRuleEntitlementV2 ruleEntitlementV2 = IRuleEntitlementV2(
+      _deployEntitlement(ims.ruleEntitlement, spaceAddress)
+    );
+
+    if (
```

Comment:
Straight copy of previous implementation with changed types, except for this extra entitlement version check here